src/api/c/CMakeLists.txt

```diff
@@ -0,0 +1,141 @@
+    ${CMAKE_CURRENT_SOURCE_DIR}/hist.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/histeq.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/histogram.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/homography.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/hsv_rgb.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/iir.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/image.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/imageio.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/imageio2.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/implicit.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/index.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/internal.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/inverse.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/join.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/lu.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/match_template.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/mean.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/meanshift.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/median.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/memory.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/moddims.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/moments.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/morph.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/nearest_neighbour.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/norm.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/orb.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/plot.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/print.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/qr.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/random.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/rank.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/reduce.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/regions.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/reorder.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/replace.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/resize.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/rgb_gray.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/rotate.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sat.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/scan.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/select.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/set.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/shift.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sift.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sobel.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/solve.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sort.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sparse.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/stdev.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/stream.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/surface.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/susan.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/svd.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/tile.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/transform.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/transform_coordinates.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/transpose.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/type_util.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/unary.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/unwrap.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/var.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/vector_field.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/where.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/window.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/wrap.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ycbcr_rgb.cpp
+    )
+
```

Comment:
Is there a way to check if a file exists in `src/api/c/` and erorr out if it is not present in `target_sources` ?